ambry-commons/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java

```diff
@@ -123,6 +124,43 @@ private void initializeDriver(String url) throws SQLException {
     }
   }
 
+  /**
+   * Enables or disables auto commit on current active {@link Connection}.
+   * @param enable {@code true} to enable auto commit and {@code false} to disable auto commit.
+   * @throws SQLException
+   */
+  public synchronized void setAutoCommit(boolean enable) throws SQLException {
+    if (activeConnection != null && activeConnection.isValid(5)) {
+      activeConnection.setAutoCommit(enable);
+    } else {
+      logger.debug("Connection to MySql DB is not available");
+    }
+  }
+
+  /**
+   * Commits transaction changes on current active {@link Connection}. This should only be used with auto commit disabled.
+   * @throws SQLException
+   */
+  public synchronized void commit() throws SQLException {
+    if (activeConnection != null && activeConnection.isValid(5)) {
+      activeConnection.commit();
+    } else {
+      logger.debug("Connection to MySql DB is not available");
+    }
+  }
+
+  /**
+   * Rollback transaction changes on current active {@link Connection}. This should only be used with auto commit disabled.
+   * @throws SQLException
+   */
+  public synchronized void rollback() throws SQLException {
+    if (activeConnection != null && activeConnection.isValid(5)) {
+      activeConnection.rollback();
+    } else {
+      logger.debug("Connection to MySql DB is not available");
```

Comment:
Minor: I feel like line 136, 148 and 160 can be `logger.warn`

Reply:
We also should not call isValid() all the time.  There was a time limit in place to prevent that.  If the connection isn't valid, the connection methods will throw an exception and that should result in cleanup.